dspy/teleprompt/gepa/gepa_utils.py

```diff
@@ -252,8 +258,12 @@ def make_reflective_dataset(self, candidate, eval_batch, components_to_update):
                         captured_trace=trace,
                     )
                     d["Feedback"] = fb["feedback"]
-                    assert fb["score"] == module_score, f"Currently, GEPA only supports feedback functions that return the same score as the module's score. However, the module-level score is {module_score} and the feedback score is {fb.score}."
-                    # d['score'] = fb.score
+                    if fb["score"] != module_score:
+                        if not self.score_mismatch_log_shown:
```

Comment:
make these just one if condition in one line, if score != module and not self. ...